framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1102,6 +1362,11 @@ public void typeProcess(TypeElement e, TreePath p) {
       // Also add possibly deferred diagnostics, which will get published back in
       // AbstractTypeProcessor.
       this.errsOnLastExit = log.nerrors;
+      printStoredMessages(p.getCompilationUnit());
+      if (!getSubcheckers().isEmpty()) {
+        // Update errsOnLastExit to reflect the errors issued.
+        this.errsOnLastExit = log.nerrors;
```

Comment:
Why is it OK to update only one `errsOnLastExit` here?  Previous code set it for every checker including subcheckers.

Reply:
The code before did not update on every checker including subcheckers.  This code is copied from `BaseTypeChecker#typeProcess` in two chunks. The first chunk is the part above the call to super.typeProcess which was copied to the top of this method: https://github.com/typetools/checker-framework/blob/723ec826cfbce855b395ed3228f5ea4df4bc720b/framework/src/main/java/org/checkerframework/common/basetype/BaseTypeChecker.java#L527-L558

The second chunk after the call to super.typeProcess was copied to the end of this method:
https://github.com/typetools/checker-framework/blob/723ec826cfbce855b395ed3228f5ea4df4bc720b/framework/src/main/java/org/checkerframework/common/basetype/BaseTypeChecker.java#L559-L565

Every subchecker eventually ends up setting`this.errsOnLastExit` when typeProcess is called on it.